chainer/functions/util/forget.py

```diff
@@ -42,13 +42,26 @@ def forward(self, inputs):
         return tuple(out.data for out in outs)
 
     def backward(self, indexes, grad_outputs):
+        # Double backprop is not allowed
+        if chainer.config.enable_backprop:
+            raise RuntimeError('double backpropagation in functions.forget is '
+                               'not allowed.')
+
         inputs = self.get_retained_inputs()
+        # Create new variables that have no creators
+        dummy_inputs = tuple([variable.Variable(inp.array) for inp in inputs])
+
         with function.force_backprop_mode():
-            outs = _call_func(self.func, inputs)
-        # Return gradients that are further backproable
-        return chainer.grad(
-            outs, inputs, grad_outputs=grad_outputs,
-            enable_double_backprop=True)
+            outs = _call_func(self.func, dummy_inputs)
+            assert len(outs) == len(grad_outputs)
+            outs = chainer.functions.identity(*outs)
+        if not isinstance(outs, tuple):
+            outs = (outs,)
```

Comment:
How about writing it as follows to avoid re-wrapping by tuple:
```py
assert len(outs) == len(grad_outputs)
if len(outs) > 1:
    outs = chainer.functions.identity(*outs)
```